zzzz_modular_occulus/code/game/machinery/bomb.dm

```diff
@@ -0,0 +1,179 @@
+				anchored = TRUE
+				if(active)
+					to_chat(user, "<span class='notice'>The bolts lock in place.</span>")
+		else
+			if(!active)
+				to_chat(user, "<span class='notice'>You wrench the bomb from the floor.</span>")
+				anchored = FALSE
+			else
+				to_chat(user, "<span class='warning'>The bolts are locked down!</span>")
+
+	else if(QUALITY_SCREW_DRIVING in I.tool_qualities)
+		open_panel = !open_panel
+		update_icon()
+		to_chat(user, "<span class='notice'>You [open_panel ? "open" : "close"] the wire panel.</span>")
+
+	else if((QUALITY_WIRE_CUTTING in I.tool_qualities) && open_panel)
+		wires.Interact(user)
+
+
+/obj/machinery/bomb/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/bomb/interact(mob/user)
+	wires.Interact(user)
+	if(!open_panel)
+		if(!active)
+			settings(user)
+		else if(anchored)
+			to_chat(user, "<span class='warning'>The bomb is bolted to the floor!</span>")
+
+/obj/machinery/bomb/proc/activate()
+	active = TRUE
+	START_PROCESSING(SSmachines, src)
+	next_beep = world.time + 10
+	detonation_timer = world.time + (timer_set * 10)
+	playsound(loc, 'sound/machines/click.ogg', 30, 1)
+
+/obj/machinery/bomb/proc/settings(mob/user)
+	var/new_timer = input(user, "Please set the timer.", "Timer", "[timer_set]") as num
+	if(in_range(src, user) && isliving(user)) //No running off and setting bombs from across the station
+		timer_set = clamp(new_timer, minimum_timer, maximum_timer)
+		loc.visible_message("<span class='notice'>[icon2html(src, viewers(src))] timer set for [timer_set] seconds.</span>")
+	if(alert(user,"Would you like to start the countdown now?",,"No","Yes") == "Yes" && in_range(src, user) && isliving(user))
+		if(defused)
+			visible_message("<span class='warning'>[icon2html(src, viewers(src))] Device error: User intervention required.</span>")
+			return
+		else
+			visible_message("<span class='danger'>[icon2html(src, viewers(loc))] [timer_set] seconds until detonation, please clear the area.</span>")
+			activate()
+			update_icon()
+			add_fingerprint(user)
+
+
+/obj/machinery/bomb/proc/try_detonate(ignore_active = FALSE)
+	if((active || ignore_active) && !defused)
+		explosion(src, 1, 5, 11, 20)
+
+
+
+//EMP CHARGE HERE
+
+
+/obj/machinery/bomb/emp
+	name = "Z-6 Ion Charge"
+	icon_state = "emp-bomb"
+	desc = "A large and menacing device known for disabling electronic equipment in a wide radius. Can be bolted down with a wrench."
+	layer = BELOW_MOB_LAYER //so people can't hide it and it's REALLY OBVIOUS
+	use_power = NO_POWER_USE	//doesn't use APC power
+	minimum_timer = 60
+	timer_set = 60
+
+/obj/machinery/bomb/emp/try_detonate(ignore_active = FALSE)
+	. = (active || ignore_active) && !defused
+	if(.)
+		empulse(get_turf(src), 14, 17, TRUE)
+	qdel(src)
+
+
+/obj/machinery/bomb/emp/update_icon()
+	icon_state = "[initial(icon_state)][active ? "-active" : "-inactive"][open_panel ? "-wires" : ""]"
```

Comment:
This is the same proc as the one in /bomb/, so it can go.

Reply:
Fixed!